engine/base/t/hash-map-test.lisp

```diff
@@ -0,0 +1,17 @@
+(defpackage :lbge.test.hash
+  (:use :cl :rove :lbge.hash))
+
+(in-package :lbge.test.hash)
+
+(deftest hash-make-equal-test
+    (ok (eq (equalp (make-hash-table) (make-hash-table))
```

Comment:
I think that the natural way of testing if `hash-equal` works properly is making a hash, adding and removing values from it, and checking its equality (and inequality) with corresponding in-place `(make-hash ...)` hashes.

This test doesn't actually test much. E.g., if `equal-hash` just always returns `t` for all arguments, this test would pass.